app/controllers/foreman_bootdisk/api/v2/disks_controller.rb

```diff
@@ -37,8 +37,13 @@ def host
           host = @disk
           if params[:full]
             return unless bootdisk_type_allowed?('full_host')
-            ForemanBootdisk::ISOGenerator.generate_full_host(host) do |iso|
-              send_file(iso, filename: "#{host.name}#{ForemanBootdisk::ISOGenerator.token_expiry(host)}.iso")
+            begin
+              ForemanBootdisk::ISOGenerator.generate_full_host(host) do |iso|
+                send_file(iso, filename: "#{host.name}#{ForemanBootdisk::ISOGenerator.token_expiry(host)}.iso")
+              end
+            rescue ::Foreman::Exception => e
```

Comment:
why do we let it try to generate the bootdisk and send it if we can check the build mode of a host beforehand? the same below

Reply:
FYI build mode is only required for full host I think (please test this but I believe this is correct).

Anyway, I do not agree with @ares here. What @domitea implemented is a typical Java-approach and it is a reasonable solution - the exception is thrown on the first line of generate_full_host.

Reply:
Granted that ideally there should be an exception class and the code should catch only specific class. Feel free to implement that if you want.

Reply:
Exceptions are slower thanks to the backtrace and should be generally used for, well, exceptional behavior. This PR updates the UI in a way it does not display the link to generate the bootdisk, so we intentionally disable it. Therefore it's not exception, it's something we know about and can easily verify before it fails (aka validation). The same thing `bootdisk_type_allowed?` check above.

I don't undestand why we'd prefer to use exception here comparing to normal program flow. This way we even need to be checking exception message and reraising it in case it's unknown code.

Reply:
That's correct, but the meaning of this is that you should not use exception for state in hotspots, like loops. This case, you would not even measure the difference. We could probably argue for some time, but I am gonna stop now - I am fine if you insist on an if statement. I do not see anything wrong with exception approach. Truth is, `host` is available in this context so why not to use it.

Reply:
I "copied" this situation from similar problem (https://github.com/theforeman/foreman_bootdisk/blob/a1d90df01320157ce5e95021ae9a633c0141006d/app/controllers/foreman_bootdisk/disks_controller.rb#L23) and I think it to use it there. Reason is simpler maintenance. The code that covers that host is not in build is already there (in `IsoGenerator`) so I made the better exception handling to say what's wrong for user. So in other words, I would like to leave as it is because I don't think it's worth make own Exception for one specific use case at one possibility. 

Also, when the host is not in built mode, I am able to make a `host image`. It makes sense because this type of image is not bonded to specific host.

Reply:
FTR I don't mind, I'm not the maintainer of this plugin :-)

Reply:
I don't mind either, I will accept whatever you just pushed Dominik. I am not maintainer of this plugin as well :-)